cmd/metal-api/internal/headscale/client.go

```diff
@@ -0,0 +1,238 @@
+package headscale
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	headscalecore "github.com/juanfont/headscale"
+	headscalev1 "github.com/juanfont/headscale/gen/go/headscale/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var (
+	apiKeyExpiration = 24 * 90 * time.Hour
+)
+
+type HeadscaleClient struct {
+	client headscalev1.HeadscaleServiceClient
+
+	address             string
+	ControlPlaneAddress string
```

Comment:
Maybe a getter method would be more precise to prevent modification from outside of this package.